embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/id/SessionIdTrackerImpl.kt

```diff
@@ -0,0 +1,41 @@
+package io.embrace.android.embracesdk.session.id
+
+import android.app.ActivityManager
+import android.os.Build
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger
+
+internal class SessionIdTrackerImpl(
+    private val activityManager: ActivityManager? = null
+) : SessionIdTracker {
+
+    @Volatile
+    private var sessionId: String? = null
+
+    override fun getActiveSessionId(): String? = sessionId
+
+    override fun setActiveSessionId(sessionId: String?, isSession: Boolean) {
+        this.sessionId = sessionId
```

Comment:
do we need to synchronize access to this?

Reply:
Ultimately yes - I want to handle that in `SessionOrchestrator` eventually.